src/frontend/js/pages/TeacherDashboardContractsLoader/TeacherDashboardContracts/index.tsx

```diff
@@ -1,144 +1,184 @@
+        (searchParams.get('signature_state') as ContractState) || ContractState.SIGNED,
+    }),
+    [],
+  );
+  const [filters, setFilters] = useState<ContractFilters>(initialFilters);
+
+  const halfSignedContractsQuery = useContracts(
+    {
+      signature_state: ContractState.LEARNER_SIGNED,
+      organization_id: filters.organization_id,
+      course_id: courseId,
+      product_id: productId,
+    },
+    { enabled: !!filters.organization_id },
+  );
+  const contractAbilities = useContractAbilities(halfSignedContractsQuery.items);
+  const canSignContract = contractAbilities.can(ContractActions.SIGN);
+  const contractToSignCount = halfSignedContractsQuery.meta?.pagination?.count ?? 0;
+  const hasContractToSign = contractToSignCount > 0;
 
   const {
     items: contracts,
     meta,
-    states: { fetching, isFetched },
-  } = useContracts(contractFilters);
-
-  const [sortModel, setSortModel] = useState<SortModel>([
-    {
-      field: 'learnerName',
-      sort: 'desc',
-    },
-    {
-      field: 'productTitle',
-      sort: 'desc',
-    },
+    states: { fetching, isFetched, error },
+  } = useContracts(
     {
-      field: 'signDate',
-      sort: 'desc',
+      ...filters,
+      course_id: courseId,
+      product_id: productId,
+      page: pagination.page,
+      page_size: PER_PAGE.teacherContractList,
     },
-  ]);
+    { enabled: !!filters.organization_id },
+  );
+
+  const rows = useMemo(() => {
+    return contracts.map((contract) => ({
+      id: contract.id,
+      learnerName: contract.order.owner_name,
+      productTitle: contract.order.product_title,
+      state: ContractHelper.getHumanReadableState(contract, ContractStatePoV.ORGANIZATION, intl),
+    }));
+  }, [contracts]);
+
+  const handleFiltersChange = (newFilters: Partial<ContractFilters>) => {
+    // Reset pagination
+    pagination.setPage(1);
+    setFilters((prevFilters) => ({ ...prevFilters, ...newFilters }));
+  };
 
-  const pagination = usePagination({
-    defaultPage: page ? parseInt(page, 10) : 1,
-    pageSize: PER_PAGE.teacherContractList,
-  });
   useEffect(() => {
-    if (isFetched) {
+    if (isFetched && meta?.pagination?.count) {
       pagination.setPagesCount(Math.ceil(meta!.pagination!.count / PER_PAGE.teacherContractList));
     }
-  }, [contracts, isFetched]);
-
-  const onPageChange = (newPage: number) => {
-    setContractFilters((currentFilters: ContractFilters) => ({ ...currentFilters, page: newPage }));
-  };
-  useEffect(() => onPageChange(pagination.page), [pagination.page]);
+  }, [meta, isFetched]);
 
-  const rows = useMemo(() => {
-    return contracts.map(({ id, student_signed_on: studentSignedOn, order }) => ({
-      id,
-      learnerName: order.owner_name,
-      productTitle: order.product_title,
-      signDate: studentSignedOn ? intl.formatDate(studentSignedOn) : studentSignedOn,
-    }));
-  }, [contracts]);
+  if (error) {
+    return <Banner message={error} type={BannerType.ERROR} rounded />;
+  }
```

Comment:
This may replace the datagrid.
if a request fail on a organization, i don't want the page layout to disapear.

Reply:
Oh yes right !

Reply:
Only the datagrid and action buttons will disappear.

<img width="800" alt="Capture d’écran 2024-01-02 à 11 48 09" src="https://github.com/openfun/richie/assets/9265241/bae2fb04-77ca-4dc5-a8e0-16167d913b6e">